File: 0792-binary-search/0792-binary-search.cs
public class Solution {
    public int Search(int[] nums, int target) 
    {
        int n = nums.Count();
        return bs(nums,0,n-1, target);
    }
    public int bs(int[] arr, int start, int end, int x)
    {
        if(start <= end)
        {
            int mid = (end + start)/2;
            if(arr[mid] == x)
                return mid;
            if(arr[mid] > x)
                return bs(arr, start, mid - 1, x);
            return bs(arr, mid + 1, end, x);
        }
        return -1;
    }
}

File: 0792-binary-search/0792-binary-search.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number}
 */
var search = function(nums, target) {
    const recur = (low, high) => {
        if(low > high) return -1;
        let mid = Math.floor((low+high)/2);
        if(nums[mid] === target) {
            return mid;
        }
        else if(nums[mid] > target) {
            return recur(low, mid - 1);
        }
        return recur(mid + 1, high);
    }
    return recur(0, nums.length - 1)
};